### PR TITLE
Moved default implementations of enforce and filter to AbstractAuthor…

### PR DESCRIPTION
**Pull request by bdmogal**
…izer

In #6073 I forgot that we had an `AbstractAuthorizer`, that could share some of the code, so not every `Authorizer` has to re-implement it.

Build: http://builds.cask.co/browse/CDAP-DUT4427

### CODE REVIEW

File: cdap-security-spi/src/main/java/co/cask/cdap/security/spi/authorization/AbstractAuthorizer.java [hunk 1]
```diff
@@ -26,14 +35,41 @@
...
+      return unfiltered;
+    }
+
+    Set<Privilege> privileges = listPrivileges(principal);
+    Set<EntityId> allowedEntities = new HashSet<>();
+    for (Privilege privilege : privileges) {
+      allowedEntities.add(privilege.getEntity());
+    }
+    Set<T> result = new HashSet<>();
+    for (T entityId : unfiltered) {
```
Comment (anwar6953) [inline cdap-security-spi/src/main/java/co/cask/cdap/security/spi/authorization/AbstractAuthorizer.java:68]: Can use Sets.intersection here?

Comment (bdmogal) [inline cdap-security-spi/src/main/java/co/cask/cdap/security/spi/authorization/AbstractAuthorizer.java:68]: nope. This is a class in cdap-security-spi, and we don't use guava there.

Comment (anwar6953) [inline cdap-security-spi/src/main/java/co/cask/cdap/security/spi/authorization/AbstractAuthorizer.java:68]: I see; ok.

**Comment by anwar6953** [comment]
lgtm

**Comment by bdmogal** [comment]
Build is green. Merging.

**Pull request merged**